proxy/proxy.go

```diff
@@ -357,6 +372,11 @@ func proxyMain() {
 		os.Exit(1)
 	}
 	proxy.serve()
+
+	// Wait for all the goroutines started by helloHandler to finish. Not
+	// stricly necessary, as the process is about to exit, but it's eye
+	// please to match wg.Add() by the corresponding wg.Wait()
```

Comment:
The purpose of wg.Add() is to be able to wait on a number of goroutines to wg.Wait() until all of them are finished. The test code does a useful wg.Wait() to ensure we end up in a clean state, but the proxy being an ever lasting daemon, the wg.Wait() here is never executed in the current state of things.